tests/bundle-test.ts

```diff
@@ -237,18 +232,18 @@ async function runAll() {
   console.log('DID workflow started')
   const { keypair, sign } = makeSigningKeypair('//Foo', 'ed25519')
 
-  const authenticationKey = keypair as NewDidVerificationKey
-  const identifier = Did.Utils.getIdentifierByKey(authenticationKey)
-  const createTx = await Did.Chain.generateCreateTxFromCreationDetails(
-    { identifier, authenticationKey },
+  const authentication = [keypair] as [NewDidVerificationKey]
+  const identifier = Did.Utils.getIdentifierByKey(authentication[0])
+  const createTx = await Did.Chain.getStoreTx(
+    { identifier, authentication },
     devFaucet.address,
     sign
   )
   await Blockchain.signAndSubmitTx(createTx, devFaucet, {
     resolveOn: Blockchain.IS_IN_BLOCK,
   })
-  const fullDid = await Did.FullDidDetails.fromChainInfo(
-    Did.Utils.getFullDidUriByKey(authenticationKey)
+  const fullDid = await Did.query(
+    Did.Utils.getFullDidUriByKey(authentication[0])
```

Comment:
Is it more accurate to name this function `getFullDidUriFromKey`?

Reply:
I’m used to thinking that I can get a kernel from a peach because the peach contains the kernel. Alternatively I can get a book by ISBN, even though the ISBN does not contain the book and only refers to it. Since the key does not, strictly speaking, contains the full DID URI, I used `By` in the name.

Reply:
But I can generate the uri _from_ the key. I can't get the book _from_ an ISBN.

Reply:
"Generate" and "get" are different verbs requiring different adverbs. I am totally with you if we were to call the function `generateFullDidUriFromKey`. We do not call it that, however, and for getting we’ll need `by`.

Reply:
I am only used to `find..by`, not really `get..by`. It sounds so much like an entity lookup from a set of items. Like `get car by serialnumber`
Additionally, since we are not just getting the uri directly from the key, `derive` would be a better word?

Reply:
While `derive` would be more semantically precise, this will differ from all the other functions in the DID namespace. As an application developer I would have to learn that this particular function starts not with `get` but with `derive`, and this would be annoying.

Reply:
Renamed in https://github.com/KILTprotocol/sdk-js/pull/585/commits/d22642372acfdc53c8c5c883e9a550a653288332